core/simplify_expression.cpp

```diff
@@ -565,7 +581,7 @@ const IR::Expression *produceSimplifiedMux(const IR::Expression *cond,
                                            const IR::Expression *falseExpression) {
     Util::ScopedTimer timer("Mux optimization");
     auto *mux = new IR::Mux(trueExpression->type, cond, trueExpression, falseExpression);
-    return mux->apply(FoldMuxConditionDown());
+    return simplify(mux->apply(FoldMuxConditionDown()));
```

Comment:
Can this be simply `return simplify(mux);` given FoldMuxConditionDown is one of the ExpressionRewriter passes?

Reply:
Will fix that